js/private/link_js_package.bzl

```diff
@@ -45,67 +45,70 @@ Can be left unspecified to allow for circular deps between `link_js_package`s.
         providers = [_LinkJsPackageInfo],
     ),
     "package": attr.string(
-        # TODO: validate that name matches in an action if src is set
-        doc = "Must match the `name` field in the `package.json` file for this package.",
-        mandatory = True,
+        doc = """The package name being linked.
+        
+Must be specified is `src` is not set. If both `src` and `package` are set, must match the JsPackageInfo package name.""",
```

Comment:
what's the use case for linking a package that doesn't have JsPackageInfo? is there an example of that already?

Reply:
inside npm_import.bzl there is a complex pattern of "reference" link_js_packages to break circular deps; I should add that not specifying a src isn't recommended outside of there since you shouldn't write 1p deps that depend on each other.

Reply:
added to docstring